test/topology_custom/test_mv_backlog.py

```diff
@@ -46,3 +46,34 @@ async def test_view_backlog_increased_after_write(manager: ManagerClient) -> Non
         assert view_backlog > v
 
     await cql.run_async(f"DROP KEYSPACE ks")
+
+# This test reproduces issues #18461 and #18783
+# In the test, we create a table and perform a write to it that fills the view update backlog.
+# After a gossip round is performed, the following write should succeed.
+@pytest.mark.asyncio
+@skip_mode('release', "error injections aren't enabled in release mode")
+async def test_gossip_same_backlog(manager: ManagerClient) -> None:
+    node_count = 2
+    # Use a higher smp to make it more likely that the writes go to a different shard than the coordinator.
+    servers = await manager.servers_add(node_count, cmdline=['--smp', '5'], config={'error_injections_at_startup': ['view_update_limit', 'update_backlog_immediately'], 'enable_tablets': True})
+    cql, hosts = await manager.get_ready_cql(servers)
+    await cql.run_async(f"CREATE KEYSPACE ks WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 1}}"
+                         "AND tablets = {'initial': 1}")
+    await cql.run_async(f"CREATE TABLE ks.tab (key int, c int, v text, PRIMARY KEY (key, c))")
+    await cql.run_async(f"CREATE MATERIALIZED VIEW ks.mv_cf_view AS SELECT * FROM ks.tab "
+                    "WHERE c IS NOT NULL and key IS NOT NULL PRIMARY KEY (c, key) ")
+    await wait_for_view(cql, 'mv_cf_view', node_count)
+
+    # Only remote updates hold on to memory, so make the update remote
+    await pin_the_only_tablet(manager, "ks", "tab", servers[0])
+    await pin_the_only_tablet(manager, "ks", "mv_cf_view", servers[1])
+
+    stmt = cql.prepare(f"INSERT INTO ks.tab (key, c, v) VALUES (?, ?, ?)")
+
+    [await manager.api.enable_injection(s.ip_addr, "never_finish_remote_view_updates", one_shot=False) for s in servers]
+    await cql.run_async(stmt, [0, 0, 240000*'a'], host=hosts[0])
+    [await manager.api.disable_injection(s.ip_addr, "never_finish_remote_view_updates") for s in servers]
+    time.sleep(1)
```

Comment:
Do we really need to waste a second here?

Reply:
We're waiting for the view update backlog gossip round to be performed, but actually, we can just rely on retries instead - I changed this in the rebase